src/async_impl/client.rs

```diff
@@ -327,6 +328,26 @@ impl ClientBuilder {
         self
     }
 
+    /// Clear all `Proxies`, so `Client` will use no proxy anymore.
+    pub fn no_proxy(mut self) -> ClientBuilder {
+        self.config.proxies.clear();
+        self
+    }
+
+    /// Add system proxy setting to the list of proxies
+    pub fn use_sys_proxy(mut self) -> ClientBuilder {
+        let proxies = get_proxies();
+        self.config.proxies.push(Proxy::custom(move |url| {
+            if proxies.contains_key(url.scheme()) {
+                return Some((*proxies.get(url.scheme()).unwrap()).clone());
+            } else {
+                return None;
+            }
```

Comment:
it is nicer to return whatever the if/else construct is giving you back ;)
```rust
            return if proxies.contains_key(url.scheme()) {
                Some((*proxies.get(url.scheme()).unwrap()).clone())
            } else {
                None
            };
```

Reply:
Since this is the last line of the callback, you could do away with `return` entirely

```rust
            if proxies.contains_key(url.scheme()) {
                Some((*proxies.get(url.scheme()).unwrap()).clone())
            } else {
                None
            };
```

Reply:
I would like to use @BO41 's comment, and I don't think throw `return` out is a good idea here, it can mess up the callback code and the outer code.  Thanks very much :) @asonix @BO41